models/db/index.go

```diff
@@ -71,10 +71,35 @@ func postgresGetNextResourceIndex(ctx context.Context, tableName string, groupID
 	return strconv.ParseInt(string(res[0]["max_index"]), 10, 64)
 }
 
+func mysqlGetNextResourceIndex(ctx context.Context, tableName string, groupID int64) (int64, error) {
+	var idx int64
+	// start a transaction to lock the mysql thread so that LAST_INSERT_ID return the correct value
+	return idx, WithTx(ctx, func(ctx context.Context) error {
+		if _, err := GetEngine(ctx).Exec(fmt.Sprintf("INSERT INTO %s (group_id, max_index) "+
+			"VALUES (?,LAST_INSERT_ID(1)) ON DUPLICATE KEY UPDATE max_index = LAST_INSERT_ID(max_index+1)",
+			// if LAST_INSERT_ID(expr) is used, next time when use LAST_INSERT_ID(), it will be given the expr value
+			tableName), groupID); err != nil {
+			return err
+		}
+
+		_, err := GetEngine(ctx).SQL("SELECT LAST_INSERT_ID()").Get(&idx)
```

Comment:
IMO no need to use `WithTx` (GetNextResourceIndex is already in transaction), no need to use `LAST_INSERT_ID` (it is not for this case, a simple `SELECT max_index` should work)